src/services/identity/SitesService.ts

```diff
@@ -175,27 +189,49 @@ class SitesService {
     }
 
     // Legacy style of commits, or if the user is not found
-    const {
-      author: { email: authorEmail },
-    } = commit
-    return authorEmail
+    return this.extractAuthorEmail(commit)
   }
 
-  async getMergeAuthorEmail(commit: GitHubCommitData) {
+  async getMergeAuthorEmail(
+    commit: GitHubCommitData,
+    sessionData: UserWithSiteSessionData
+  ) {
     const {
       author: { name: authorName },
     } = commit
+    const { siteName } = sessionData
+
+    if (!authorName.startsWith("isomergithub")) {
+      // Legacy style of commits, or if the user is not found
+      return this.extractAuthorEmail(commit)
+    }
```

Comment:
should we attempt to parse the message rather than checking the prefix of the author name?

Reply:
Hmm in this case, there would not be a message. It is likely that it might be something like "Merged pull request #... from staging". So we would want to extract the email address of the commit author instead.